File: com/satyajit/demo/HelloWorld.java
package com.satyajit.demo;
import java.util.Arrays;

public class HelloWorld {

    public static void main(String[] args) {
        System.out.println("Hello world!");
        HelloWorld hw = new HelloWorld();
        hw.print(new int[]{1,4, 2, 3});
    }

    public void print(int[] arr) {
        Arrays.sort(arr);
        System.out.println(Arrays.toString(arr));
    }
}

File: helloworld.r

Hello_World_Java <- function() {
  hello_world <- J("com/satyajit/demo/HelloWorld")
  hello <- .jnew(hello_world)
  arr <- .jarray(as.integer(c(3, 5, 2, 1, 6)))
  .jcall(hello, "V", "print", arr)
}


